github/checks.go

```diff
@@ -0,0 +1,428 @@
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req.Header.Set("Accept", mediaTypeCheckRunsPreview)
+
+	var checkRunResults *ListCheckRunsResults
+	resp, err := s.client.Do(ctx, req, &checkRunResults)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return checkRunResults, resp, nil
+}
+
+// ListCheckSuiteOptions Optional parameters to list check suites.
+type ListCheckSuiteOptions struct {
+	CheckName *string `url:"check_name,omitempty"` //Filters checks suites by the name of the check run``.
+	Filter    *string `url:"filter,omitempty"`     //Filters check suites by GitHub App id.
+
+	ListOptions
+}
+
+// ListCheckSuiteResults represents the result of a check run list.
+type ListCheckSuiteResults struct {
+	Total       *int         `json:"total_count,omitempty"`
+	CheckSuites []CheckSuite `json:"check_suites,omitempty"`
+}
+
+// ListCheckSuitesForRef List check suite for a specific ref.
+//
+// GitHub API docs: https://developer.github.com/v3/checks/suites/#list-check-suites-for-a-specific-ref
+func (s *ChecksService) ListCheckSuitesForRef(ctx context.Context, owner string, repo string, ref string, opt *ListCheckSuiteOptions) (*ListCheckSuiteResults, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/commits/%v/check-suites", owner, repo, ref)
+	u, err := addOptions(u, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req.Header.Set("Accept", mediaTypeCheckRunsPreview)
+
+	var checkSuiteResults *ListCheckSuiteResults
+	resp, err := s.client.Do(ctx, req, &checkSuiteResults)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return checkSuiteResults, resp, nil
+}
+
+// AutoTriggerCheck Enables or disables automatic creation of CheckSuite events upon pushes to the repository.
+type AutoTriggerCheck struct {
+	AppID   *int64 `json:"app_id,omitempty"`  // The id of the GitHub App. (Required.)
+	Setting *bool  `json:"setting,omitempty"` // Set to true to enable automatic creation of CheckSuite events upon pushes to the repository, or false to disable them. Default: true (Required.)
+}
+
+// CheckSuitePreferenceOptions Set options for check suite preferences for a repository.
+type CheckSuitePreferenceOptions struct {
+	PreferenceList *PreferenceList `json:"auto_trigger_checks,omitempty"` // An array of auto trigger checks that can be set for a check suite in a repository.
+}
+
+// CheckSuitePreferenceResults Represents the results of the preference set operation.
+type CheckSuitePreferenceResults struct {
+	Preferences *PreferenceList `json:"preferences,omitempty"`
+	Repository  *Repository     `json:"repository,omitempty"`
+}
+
+// PreferenceList Represents a list of auto trigger checks for  repository
+type PreferenceList struct {
+	AutoTriggerChecks []AutoTriggerCheck `json:"auto_trigger_checks,omitempty"` // An array of auto trigger checks that can be set for a check suite in a repository.
```

Comment:
`[]*AutoTriggerCheck`

Also, maybe change `array` to `slice` to fit Go nomenclature better (here and elsewhere).